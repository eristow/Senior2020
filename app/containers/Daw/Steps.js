import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { makeSelectStepState } from './selectors';
import Step from './Step';

const Wrapper = styled.div`
  display: flex;
  flex: 1;
`;

export function Steps({ name, num, stepState }) {
  return (
    <Wrapper>
      {stepState[num].map((s, i) => (
        <Step
          on={s !== 0}
          doubled={s === 2}
          index={i}
          // eslint-disable-next-line react/no-array-index-key
          key={`${name} ${i}`}
          num={num}
        />
      ))}
    </Wrapper>
  );
}

Steps.propTypes = {
  name: PropTypes.string,
  num: PropTypes.number,
  stepState: PropTypes.array,
};

const mapStateToProps = createStructuredSelector({
  stepState: makeSelectStepState(),
});

const withConnect = connect(mapStateToProps);

export default compose(withConnect)(Steps);
