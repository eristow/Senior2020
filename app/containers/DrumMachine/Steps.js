import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { makeSelectStepState } from './selectors';
import Step from './Step';

const Wrapper = styled.div`
  display: flex;
  flex: 1;
`;

export function Steps({ name, stepState }) {
  return (
    <Wrapper>
      {stepState[name].map((s, i) => (
        <Step
          on={s !== 0}
          doubled={s === 2}
          index={i}
          // eslint-disable-next-line react/no-array-index-key
          key={`${name} ${i}`}
          name={name}
        />
      ))}
    </Wrapper>
  );
}

Steps.propTypes = {
  name: PropTypes.string,
  stepState: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  stepState: makeSelectStepState(),
});

const withConnect = connect(mapStateToProps);

export default compose(withConnect)(Steps);
