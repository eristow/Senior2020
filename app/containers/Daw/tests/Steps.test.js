import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';

import { Steps } from '../Steps';

describe('<Steps />', () => {
  let store;
  const stepState = [
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  ];
  const num = 0;

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();

    render(
      <Provider store={store}>
        <Steps dispatch={dispatch} stepState={stepState} num={num} />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <Steps stepState={stepState} num={num} />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
