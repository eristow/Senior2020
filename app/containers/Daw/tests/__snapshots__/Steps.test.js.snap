// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Steps /> Should render and match the snapshot 1`] = `
<div
  class="Steps__Wrapper-sc-8e49i7-0 iDvuHY"
>
  <button
    class="Step__StepButton-sc-1gy89br-0 buOaSX"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 buOaSX"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 buOaSX"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 buOaSX"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 iNYnBq"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 iNYnBq"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 iNYnBq"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 iNYnBq"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 buOaSX"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 buOaSX"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 buOaSX"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 buOaSX"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 iNYnBq"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 iNYnBq"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 iNYnBq"
  />
  <button
    class="Step__StepButton-sc-1gy89br-0 iNYnBq"
  />
</div>
`;
